types/duration.go

```diff
@@ -0,0 +1,258 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+	"unicode"
+)
+
+const (
+	convertDays    int64 = 86400000
+	convertHours   int64 = 3600000
+	convertMinutes int64 = 60000
+	convertSeconds int64 = 1000
+)
+
+var unitToMillis = map[string]int64{
+	"d":  convertDays,
+	"h":  convertHours,
+	"m":  convertMinutes,
+	"s":  convertSeconds,
+	"ms": 1,
+}
+
+var unitOrder = []string{"d", "h", "m", "s", "ms"}
+
+// A Duration is a value representing a span of time in milliseconds.
+type Duration struct {
+	Value int64
+}
+
+// UnsafeDuration creates a duration via unsafe conversion from int, int64, float64
+func UnsafeDuration[T int | int64 | float64](v T) Duration {
```

Comment:
Maybe it should multiply by 1000 or something?  And the comment should indicate what it takes in, e.g. seconds?

Reply:
I'd probably just drop this function and only include FromStdDuration.